packages/@ember/debug/tests/main_test.js

```diff
@@ -273,6 +328,77 @@ moduleFor(
       );
     }
 
+    ['@test deprecate without options.for triggers an assertion'](assert) {
+      assert.expect(1);
+
+      assert.throws(
+        () => deprecate('message1', false, { id: 'test', until: 'forever' }),
+        new RegExp(missingOptionsForDeprecation('test')),
+        'proper assertion is triggered when options.for is missing'
+      );
+    }
```

Comment:
Unlike the other options, these don't throw (they issue deprecations).

```suggestion
    ['@test deprecate without options.for triggers a deprecation'](assert) {
      assert.expect(1);

      expectDeprecation(
        () => deprecate('message1', false, { id: 'test', until: 'forever' }),
        new RegExp(missingOptionsForDeprecation('test')),
        'deprecation is triggered when options.for is missing'
      );
    }
```

Reply:
hmm, 🤔 , I assumed it was becuase of `ENV.RAISE_ON_DEPRECATION = true` being set in the `beforeEach`?

Do you want to change the actual code after changing this test case also?

Reply:
changing this test does make it keep passing, but then it seems like we should also change `ENV.RAISE_ON_DEPRECATION = false` in these tests, which leave some ground uncovered when the flag is true.

Reply:
chatted offline. using `expectDeprecation` is a better pattern, because it handles both the dev and prod builds (in prod, deprecations don't run, so the test would do the right thing there). in this case, it doesn't matter because this test never runs against the prod build (b/c it's a debug specific test), but it's better to follow the same pattern as the rest of the test suite.

Reply:
@rwjblue I had to revert this. `expectDeprecation` stubs out the `deprecate()` function so it never calls it the second time :/ Not sure what the best path forward is.

Reply:
Ahh. Gotcha. I think this is probably fine then. It seems that we can't satisfy the consistency constraint without a bunch of hoop jumping, and this package's tests aren't even ran in prod builds (where `expectDeprecation` actually is required).